flink-tests/src/test/java/org/apache/flink/test/windowing/sessionwindows/SessionWindowITCase.java

```diff
@@ -114,8 +115,8 @@ private void runTest(
 			windowedStream = windowedStream.allowedLateness(Time.milliseconds(ALLOWED_LATENESS_MS));
 		}
 
-		if (!PURGE_WINDOW_ON_FIRE) {
-			windowedStream = windowedStream.trigger(new NonPurgingEventTimeTriggerWrapper());
+		if (PURGE_WINDOW_ON_FIRE) {
+			windowedStream = windowedStream.trigger(PurgingTrigger.of(EventTimeTrigger.create()));
 		}
```

Comment:
I do not know if this behavior existed before, but now if you set the PURGE_WINDOW_ON_FIRE to true, now the test fails.